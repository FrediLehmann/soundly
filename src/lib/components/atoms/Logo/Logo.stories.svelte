<script>
  import { Meta, Story, Template } from '@storybook/addon-svelte-csf';

  import Logo from './Logo.svelte';
</script>

<Meta
  title="Atoms/Logo"
  component={Logo}
  argTypes={{
    color: {
      control: { type: 'select' },
      options: ['black', 'white', 'primary'],
      table: {
        defaultValue: {
          summary: 'primary'
        },
        type: {
          summary: 'Which color the logo should be'
        }
      }
    },
    size: {
      control: { type: 'select' },
      options: ['small', 'medium', 'big'],
      table: {
        defaultValue: {
          summary: 'small'
        },
        type: {
          summary: 'The size of the logo.'
        }
      }
    },
    href: {
      control: { type: 'string' },
      table: {
        type: {
          summary: 'Link somewehere on click'
        }
      }
    }
  }} />

<Template let:args>
  <div class="container">
    <Logo {...args} size="small" />
    <Logo {...args} size="medium" />
    <Logo {...args} size="big" />
  </div>
</Template>

<Story name="Black" args={{ color: 'black' }} />

<Story name="White" args={{ color: 'white' }} />

<Story name="Primary" args={{ color: 'primary' }} />

<style>
  .container {
    background-color: lightgray;
    padding: 20px;

    display: flex;
    flex-direction: column;
  }
</style>
