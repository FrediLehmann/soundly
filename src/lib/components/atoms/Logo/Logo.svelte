<script lang="ts" context="module">
  import { goto } from '$app/navigation';

  import LogoFullBlackPNG from './assets/logo_full_black.png';
  import LogoFullBlackWEBP from './assets/logo_full_black.webp';
  import LogoFullWhitePNG from './assets/logo_full_white.png';
  import LogoFullWhiteWEBP from './assets/logo_full_white.webp';
  import LogoFullPrimaryPNG from './assets/logo_full_primary.png';
  import LogoFullPrimaryWEBP from './assets/logo_full_primary.webp';

  const images = {
    png: {
      black: LogoFullBlackPNG,
      white: LogoFullWhitePNG,
      primary: LogoFullPrimaryPNG
    },
    webp: {
      black: LogoFullBlackWEBP,
      white: LogoFullWhiteWEBP,
      primary: LogoFullPrimaryWEBP
    }
  };
</script>

<script lang="ts">
  export let color: 'black' | 'white' | 'primary' = 'primary';
  export let size: 'small' | 'medium' | 'big' = 'small';
  export let href: string;
</script>

<picture data-size={size} on:click={() => goto(href)}>
  <source srcset={images.webp[color]} />
  <img alt="Noiceless Logo" src={images.png[color]} />
</picture>

<style>
  img {
    aspect-ratio: 8.5/1;
  }

  picture:hover {
    cursor: pointer;
  }

  picture[data-size='small'] > img {
    width: 8rem;
  }

  picture[data-size='medium'] > img {
    width: 12rem;
  }

  picture[data-size='big'] > img {
    width: 15rem;
  }
</style>
