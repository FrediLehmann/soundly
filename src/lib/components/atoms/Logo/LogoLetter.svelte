<script lang="ts" context="module">
  import { goto } from '$app/navigation';

  import LogoLetterBlackPNG from './assets/logo_letter_black.png';
  import LogoLetterBlackWEBP from './assets/logo_letter_black.webp';
  import LogoLetterWhitePNG from './assets/logo_letter_white.png';
  import LogoLetterWhiteWEBP from './assets/logo_letter_white.webp';
  import LogoLetterPrimaryPNG from './assets/logo_letter_primary.png';
  import LogoLetterPrimaryWEBP from './assets/logo_letter_primary.webp';

  const images = {
    png: {
      black: LogoLetterBlackPNG,
      white: LogoLetterWhitePNG,
      primary: LogoLetterPrimaryPNG
    },
    webp: {
      black: LogoLetterBlackWEBP,
      white: LogoLetterWhiteWEBP,
      primary: LogoLetterPrimaryWEBP
    }
  };
</script>

<script lang="ts">
  export let color: 'black' | 'white' | 'primary' = 'primary';
  export let size: 'small' | 'medium' | 'big' = 'small';
  export let href: string;
</script>

<picture data-size={size} on:click={() => goto(href)}>
  <source srcset={images.webp[color]} />
  <img alt="Noiceless Logo" src={images.png[color]} />
</picture>

<style>
  img {
    aspect-ratio: 1/1;
  }

  picture:hover {
    cursor: pointer;
  }

  picture[data-size='small'] > img {
    height: 1.25rem;
  }

  picture[data-size='medium'] > img {
    height: 2.5rem;
  }

  picture[data-size='big'] > img {
    height: 4rem;
  }
</style>
