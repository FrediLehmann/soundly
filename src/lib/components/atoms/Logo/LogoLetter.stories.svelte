<script>
  import { Meta, Story, Template } from '@storybook/addon-svelte-csf';

  import LogoLetter from './LogoLetter.svelte';
</script>

<Meta
  title="Atoms/LogoLetter"
  component={LogoLetter}
  argTypes={{
    color: {
      control: { type: 'select' },
      options: ['black', 'white', 'primary'],
      table: {
        defaultValue: {
          summary: 'primary'
        },
        type: {
          summary: 'Which color the logo should be'
        }
      }
    },
    size: {
      control: { type: 'select' },
      options: ['small', 'medium', 'big'],
      table: {
        defaultValue: {
          summary: 'small'
        },
        type: {
          summary: 'The size of the logo.'
        }
      }
    }
  }} />

<Template let:args>
  <div class="container">
    <LogoLetter {...args} size="small" />
    <LogoLetter {...args} size="medium" />
    <LogoLetter {...args} size="big" />
  </div>
</Template>

<Story name="Black" args={{ color: 'black' }} />

<Story name="White" args={{ color: 'white' }} />

<Story name="Primary" args={{ color: 'primary' }} />

<style>
  .container {
    background-color: lightgray;
    padding: 20px;

    display: flex;
    flex-direction: column;
  }
</style>
