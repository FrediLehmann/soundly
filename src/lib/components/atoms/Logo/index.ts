import Logo from './Logo.svelte';
import LogoLetter from './LogoLetter.svelte';

export { Logo, LogoLetter };
