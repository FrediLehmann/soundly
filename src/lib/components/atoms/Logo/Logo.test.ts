/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Logo from './Logo.svelte';

describe('<Logo /> component', () => {
  test('renders correctly', () => {
    const { component } = render(Logo);
    expect(component).toMatchSnapshot();
  });
});
