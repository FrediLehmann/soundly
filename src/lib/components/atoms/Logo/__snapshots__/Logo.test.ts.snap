// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> component renders correctly 1`] = `
Logo {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "primary",
      "small",
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "color": 0,
      "size": 1,
    },
    "root": <div>
      <picture
        class="svelte-4dorz4"
        data-size="small"
      >
        <source />
         
        <img
          alt="Noiceless Logo"
          class="svelte-4dorz4"
        />
      </picture>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
